options.go

```diff
@@ -234,6 +236,21 @@ func WithIAMAuthN() Option {
 	}
 }
 
+// InstanceConnectionNameResolver resolves  This allows an application to replace the default
+// DNSInstanceConnectionNameResolver with a custom implementation.
+type InstanceConnectionNameResolver interface {
```

Comment:
This should probably go in the `instance` package next to the code that parses instance connection names. And that way we could just call it `Resolver`.

Reply:
Yes. See review comment.